pkg/asset/installconfig/vsphere/permission_test.go

```diff
@@ -18,82 +23,149 @@ func buildPermissionGroup(authManagerMock *mock.MockAuthManager,
 	username string,
 	group PermissionGroupDefinition,
 	groupName permissionGroup,
-	overrideGroup *permissionGroup) {
+	overrideGroup *permissionGroup,
+	permissionToExcludeSet []string) {
 	permissionToApply := group.Permissions
+	filteredPermissionsToApply := permissionToApply
 	if overrideGroup != nil && *overrideGroup == groupName {
-		permissionToApply = permissionToApply[:len(permissionToApply)-1]
+		filteredPermissionsToApply = []string{}
+		for _, permission := range group.Permissions {
+			include := true
+			for _, permissionToExclude := range permissionToExcludeSet {
+				if permission == permissionToExclude {
+					include = false
+					break
+				}
+			}
+			if include {
+				filteredPermissionsToApply = append(filteredPermissionsToApply, permission)
+			}
+		}
```

Comment:
Maybe this can be simplified by using https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets ?